src/Microsoft.Extensions.Logging.Console/Internal/ConsoleLoggerProcessor.cs

```diff
@@ -54,10 +57,14 @@ internal virtual void WriteMessage(LogMessageEntry message)
 
         private void ProcessLogQueue()
         {
-            foreach (var message in _messageQueue.GetConsumingEnumerable())
+            try
             {
-                WriteMessage(message);
+                foreach (var message in _messageQueue.GetConsumingEnumerable())
+                {
+                    WriteMessage(message);
+                }
             }
+            catch { }
```

Comment:
If this ever happens it'll be a pain to figure out what went wrong with your logging.

Reply:
Actually on second thought, I think we need to close the _messageQueue here as well. Otherwise it can just back up until it reaches the max and then the application will just block trying to log.

Reply:
Just don't forget to wrap that code into try/catch too.